packages/dropdownMenu/tests/DropdownMenu.test.tsx

```diff
@@ -50,10 +47,10 @@ describe("Dropdown", () => {
       </DropdownMenu>
     );
 
-    expect(toJson(component)).toMatchSnapshot();
+    expect(baseElement).toMatchSnapshot();
```

Comment:
I found baseElement to be more robust that asFragment.

After rebuilding the snapshots, and changing the menu item text, I found the test to still pass.

With baseElement, changes to the menu item text failed the test, as I expected.

Reply:
Do you have more information on why `baseElement` would be preferred here?

I read through this Kent C. Dodds's article recently and he mentioned this, 
> The benefit of using screen is you no longer need to keep the render call destructure up-to-date as you add/remove the queries you need. You only need to type screen. and let your editor's magic autocomplete take care of the rest.
> 
> The only exception to this is if you're setting the container or baseElement which you probably should avoid doing (I honestly can't think of a legitimate use case for those options anymore and they only exist for historical reasons at this point).
This is the article - https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen 

And from the [RTL docs](https://testing-library.com/docs/react-testing-library/api/#baseelement-1), 

> Note: the queries returned by the render looks into baseElement, so you can use queries to test your portal component without the baseElement.

We're definitely open to learning new ideas and improvements. Would love to hear your thoughts!

Reply:
For demonstration purposes, I've changed the `renders an open dropdown with multiple sections` test and updated the snapshots
<img width="1072" alt="asFragment before text edit" src="https://user-images.githubusercontent.com/8443215/195621028-80b6b8ec-22b0-4615-8590-730734096588.png">

This is me changing the Trigger text. It fails, so we're good here.
<img width="1028" alt="asFragment - trigger text changed" src="https://user-images.githubusercontent.com/8443215/195621042-06ee3687-60ac-4a7c-a250-9432d27f4655.png">

I originally modified the text in the menu items and the changes weren't caught.

But I just found this a moment. This is really interesting.
<img width="1028" alt="Screen Shot 2022-10-14 at 1 22 33 am" src="https://user-images.githubusercontent.com/8443215/195623299-ec032ef9-3125-44ce-979b-0b0e8e7711dd.png">


### How I arrived at baseElement
<img width="947" alt="Screen Shot 2022-10-14 at 1 11 40 am" src="https://user-images.githubusercontent.com/8443215/195621242-d1180b37-bf62-45af-90cd-2000cec52d05.png">

This is going to sound funny. But I googled "snapshot testing react testing library", landed on the docs, then searched for "snapshot". I didn't even read the docs 🤣.

I ran the same checks as above and baseElement catches these changes.

Reply:
I just checked and the DropdownMenu definitely using a portal so the Menu Contents are being rendered in a different subtree.

It's definitely an edge case. This has been a really interested rabbit-hole to dive into.

Reply:
Love the screenshots. Thank you. That makes sense to me in that we can reach for either asFragment, baseElement, or screen depending on the complexity and the type of element. Really interesting, and I appreciate you sharing your findings!

Reply:
This is very informative. Thank you for the thought out comments!